src/Symfony/Component/Form/Form.php

```diff
@@ -507,6 +508,10 @@ public function handleRequest($request = null)
      */
     public function submit($submittedData, $clearMissing = true)
     {
+        if ($submittedData instanceof Request) {
+            return $this->handleRequest($submittedData);
```

Comment:
also you need to skip the following code in this case because the request handler itself also calls `submit`. so otherwise the form would be submitted twice and and AlreadySubmittedException is thrown.
have you actually tried to use the fix for your issue? it cannot work in its current form AFAIK.